spec/promoted/ruby/client_spec.rb

```diff
@@ -692,29 +692,29 @@
   end
 
   context "prepare_for_logging when user defined method is passed" do
-    let!(:to_compact_metrics_insertion_func) do
-      Proc.new do |insertion|
-        insertion[:properties].delete(:invites_required)
-        insertion[:properties].delete(:should_discount_addons)
-        insertion[:properties].delete(:total_uses)
-        insertion[:properties].delete(:is_archived)
-        insertion
+    let!(:to_compact_metrics_properties_func) do
+      Proc.new do |properties|
+        properties[:struct].delete(:invites_required)
```

Comment:
The need to deal with :struct makes this a bit more error-prone that I'd hope. Perhaps we should just pass what's _under_ struct (make that the method contract) and fix it up at the call site. @danbosnichill wdyt?

Reply:
I'm open in the short-term.  It's hard to say until we figure out whatever more efficient format we use (e.g. protobuf and grpc).  We can always change that later though.  I probably wouldn't call it properties if we're referring to properties.struct.

Reply:
I will go with this for now unless you have a strong opinion. It's a little less troublesome since I'm passing the method a deep copy with the rationale that there are no cases where a shallow copy would be correct in practice for a non-trivial compact function.